### ISSUE TITLE
Infinite loop for Bytes2String

### ISSUE THREAD

**Issue by virgil-serbanuta** [runtimeverification/llvm-backend#847]
As far as I can tell, `Bytes2String(b"\xa9OSt\xfc\xe5\xed\xbc\x8e*\x86\x97\xc1S1g~n\xbf\x0b\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00")` enters an infinite loop when running with the LLVM backend.

Version:
```
$ kompile --version
K version:    v6.0.119-0-g015e5720dc-dirty
Build date:   Vi sep 29 20:18:24 EEST 2023
```

Input files:
a.k:
```
module A-SYNTAX
    imports INT-SYNTAX

    syntax KItem  ::= a(Int)
endmodule

module A
    imports BYTES
    imports A-SYNTAX

    rule a(_) => Bytes2String(b"\xa9OSt\xfc\xe5\xed\xbc\x8e*\x86\x97\xc1S1g~n\xbf\x0b\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00")
endmodule
```
start:
```
a(0)
```
Command line:
```
$ kompile a.k
$ krun start
```

**Comment by Robertorosmaninho** [comment]
Problems occur only on `llvm-kompile-testing`. Needs investigating. Unicode changes will fix this.

**Comment by Robertorosmaninho** [comment]
blocked on Unicode.

**Comment by Scott-Guest** [comment]
The infinite loop was introduced by #828, and it was fixed by reverting those changes in #845

**Comment by Scott-Guest** [comment]
While the actual issue submitted here is fixed, there are further issues with this test case due to the conflation between Bytes and Unicode strings. See this PR #849. This will be fixed when #828 is un-reverted.
